lib/node-token-validation/src/cache/NodeCacheManager.ts

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AuthorityMetadataEntity, CacheManager, ICrypto, Logger, ValidCacheType } from "@azure/msal-common";
+import { CacheKVStore } from "./CacheTypes";
+
+/**
+ * This class class implements storage for node-token-validation.
+ */
+export class NodeCacheManager extends CacheManager {
+    private logger: Logger;
+    private cache: CacheKVStore = {};
+    private changeEmitters: Array<Function> = [];
+
+    constructor(logger: Logger, clientId: string, cryptoImpl: ICrypto) {
+        super(clientId, cryptoImpl);
+        this.logger = logger;
+    }
+
+    /**
+     * Queue up callbacks
+     * @param func A callback function for cache change indication
+     */
+    registerChangeEmitter(func: () => void): void {
+        this.changeEmitters.push(func);
+    }
+    
+    /**
+     * Invoke the callback when cache changes
+     */
+    emitChange(): void {
+        this.changeEmitters.forEach(func => func.call(null));
+    }
+
+    /**
+     * Gets the current cache key-value store
+     * @returns {CacheKVStore} Key Value store for in-memory storage
+     */
+    getCache(): CacheKVStore {
```

Comment:
Some of this is also implemented in `msal-node` did you consider moving this to `msal-common` so you can share code?

Reply:
After talking with @sameerag, it seems that having this code in `msal-node` as opposed to `msal-common` was a deliberate choice, so that each platform retains control over storage processes in case it changes. It does mean some of the code would be duplicated in `node-token-validation` since it's also a node library, but the priority is control over code optimization.

Reply:
That may have been the case when it was just browser and node but we may want to revisit that decision with the introduction of another node based library. Are there any known existing or future scenarios where the cache implementation may differ that could justify writing it twice? Potentially a question for design review?